crates/c-api/src/ext.rs

```diff
@@ -0,0 +1,99 @@
+//! This file defines the extern "C" API extension, which are specific
+//! to the wasmtime implementation.
+
+use wasmtime::{Config, OptLevel, Strategy};
+
+#[repr(u8)]
+#[derive(Clone)]
+pub enum wasmtime_strategy_t {
+    WASMTIME_STRATEGY_AUTO,
+    WASMTIME_STRATEGY_CRANELIFT,
+    WASMTIME_STRATEGY_LIGHTBEAM,
+}
+
+#[repr(u8)]
+#[derive(Clone)]
+pub enum wasmtime_opt_level_t {
+    WASMTIME_OPT_LEVEL_NONE,
+    WASMTIME_OPT_LEVEL_SPEED,
+    WASMTIME_OPT_LEVEL_SPEED_AND_SIZE,
+}
+
+#[repr(C)]
+#[derive(Clone)]
+pub struct wasmtime_config_t {
```

Comment:
I am leaning towards leaving this type as `wasm_config_t` in `lib.rs` as this would make the signatures for `wasm_config_new` and `wasm_engine_new_with_config` match what's in the C API.  This makes it easier to search for what's implemented from the C header, in my opinion.

What do you think?